### PR TITLE
fix: harden Node SDK workflows, worker, dashboard

### PR DESCRIPTION
**Pull request by pratyush618**
## Summary
- Fixes all 18 findings from the internal Node SDK review (scan-heavy sync I/O blocking the event loop, workflow-tracker hangs, crash vectors, dashboard contract mismatches)
- Two of the workflow-tracker fixes are mirrored into the Python tracker (same bugs existed there)

## Changes
- **Async offload (perf)**: ~11 scan-heavy napi queue methods (stats*, listJobs, deadLetters*, purge*, listWorkers, getMetrics, getJobErrors) moved to `async fn` + `spawn_blocking`; JS wrappers now return Promises. Dashboard, REST, CLI, contrib, and tests updated. README notes the remaining sync point-op caveat.
- **Workflow tracker (Node + Python)**: gate resolution now routes through `settle()` so saga compensation and sub-workflow parent resolution fire; static successors of failed nodes are skipped instead of blocked forever on managed runs; fan-out sibling successors are evaluated alongside fan-in; one tracker shared per queue so manual gate resolution clears the worker's gate timers; workflow-advance failures logged at error level.
- **Worker/runtime hardening**: crossbeam result send moved off the shared tokio runtime onto the blocking pool; `catch_unwind` around `handle_result` with logged channel-closed sends; worker lifecycle I/O (register/heartbeat/unregister) moved to the blocking pool; async listener/hook rejections caught instead of crashing the process; dashboard server gets an `error` listener; job payload moved instead of cloned into invocations.
- **Dashboard/REST contract**: webhook deliveries reshaped to match the SPA contract (paged `{items,total,limit,offset}`, status enum, payload/response/latency/timestamps); `cache_hit`/`compensated` counted as terminal in stats; REST result endpoint clamps client-supplied timeout (DoS guard); `WebhookValidationError` now derives from `TaskitoError`; worker resource rebuilt on re-registration.
- **Perf**: cursor-based task-log streaming — new `get_task_logs_after` storage method (traits + diesel_common + all 3 backends + delegate + Node binding + contract test); `stream()` tracks a UUIDv7 cursor instead of rescanning full history each poll.

## Breaking changes
- Scan-heavy `Queue` methods (stats, listJobs, deadLetters*, purge*, listWorkers, getMetrics, getJobErrors) now return Promises — call sites need `await`.
- The webhook `Delivery` type is reshaped (status enum + richer fields) to match the dashboard contract.

## Test plan
- [x] `cargo test --workspace` green
- [x] `cargo check --features postgres` / `--features redis` green
- [x] `cargo clippy` green
- [x] Node vitest 192/192 (44 files, includes new regression tests)
- [x] Python `tests/workflows` 100/100, including 2 new regression tests
- [x] `ruff` + `mypy` clean

## Summary by CodeRabbit

* **New Features**
  * Added cursor-based task log retrieval (`afterId`) across storage, queue/SDK, and dashboard/queue APIs.
  * Improved job partial streaming and webhook delivery logs with cursor/pagination support.
* **Bug Fixes**
  * Moved scan-heavy inspection/admin calls off the event loop to async/threaded execution.
  * Hardened workers and async event/hook handling against panics and rejection crashes; improved REST timeout clamping.
  * Refined workflow run tracking and edge-case behavior.
* **Documentation**
  * Updated Node SDK and dashboard/operations docs for the new async patterns and webhook delivery logs.
* **Tests**
  * Added integration coverage for cursor-based task log retrieval behavior.

### CODE REVIEW

File: crates/taskito-core/src/storage/redis_backend/logs.rs [hunk 1]
```diff
@@ -93,6 +93,21 @@ impl RedisStorage {
...
+    ) -> Result<Vec<TaskLogRow>> {
+        let mut rows = self.get_task_logs(job_id)?;
+        if let Some(cursor) = after_id {
+            rows.retain(|row| row.id.as_str() > cursor);
+        }
+        // UUIDv7 ids sort by creation time, so id order == time order.
+        rows.sort_by(|a, b| a.id.cmp(&b.id));
+        Ok(rows)
+    }
+
```
[the remark this answers is not in the document]
Comment (pratyush618) [reply crates/taskito-core/src/storage/redis_backend/logs.rs:129]: Fixed in 9070e991 — ids are filtered against the cursor after ZRANGEBYSCORE, before any GET, so old entries cost one range read instead of a value fetch each.

File: sdks/node/src/queue.ts [hunk 2]
```diff
@@ -408,59 +428,59 @@ export class Queue<TTasks extends TaskMap = TaskMap> {
...
-  getJobErrors(id: string): JobError[] {
+  getJobErrors(id: string): Promise<JobError[]> {
     return this.native.getJobErrors(id);
   }
 
   /** Per-execution task metrics within the last `sinceMs` milliseconds. */
-  getMetrics(sinceMs: number, task?: string): Metric[] {
+  getMetrics(sinceMs: number, task?: string): Promise<Metric[]> {
     return this.native.getMetrics(task ?? null, sinceMs);
   }
```
[the remark this answers is not in the document]
Comment (pratyush618) [reply sdks/node/src/queue.ts]: Fixed in 501bc06b — JSDoc now states sinceMs is a Unix-epoch-ms cutoff.
